### PR TITLE
14 new + 1 modified GHSA-based advisories

### PR DESCRIPTION
**Pull request by jasnow**
14 new + 1 modified GHSA-based (2006-2009) advisories

New:
  * 14 "rails" gem advisories
  * 1 "thin" gem advisories
  
Modified:
  * 1 "ruby" advisories (associated with webrick)

### CODE REVIEW

File: gems/rails/CVE-2009-3009.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+gem: rails
+cve: 2009-3009
+osvdb: 57666
+ghsa: 8qrh-h9m2-5fvf
+url: http://weblog.rubyonrails.org/2009/9/4/xss-vulnerability-in-ruby-on-rails
```
Comment (postmodern) [inline gems/rails/CVE-2009-3009.yml:6]: Same here. The patches reference `activesupport` and `actionpack`. Probably OK to copy this file into `gems/activesupport` and `gems/actionpack`.

File: gems/rails/CVE-2009-3086.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+gem: rails
+cve: 2009-3086
+ghsa: fg9w-g6m4-557j
+url: http://weblog.rubyonrails.org/2009/9/4/timing-weakness-in-ruby-on-rails
```
Comment (postmodern) [inline gems/rails/CVE-2009-3086.yml:5]: Patches reference `actionpack`.

File: gems/rails/CVE-2009-4214.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+---
+gem: rails
+cve: 2009-4214
+ghsa: 9p3v-wf2w-v29c
+url: http://weblog.rubyonrails.org/2009/11/30/ruby-on-rails-2-3-5-released
```
Comment (postmodern) [inline gems/rails/CVE-2009-4214.yml:5]: Patches reference `actionpack`.

File: gems/rails/CVE-2009-2422.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  - ">= 2.3.3"
+related:
+  url:
+    - https://nvd.nist.gov/vuln/detail/CVE-2009-2422
+    - http://weblog.rubyonrails.org/2009/6/3/security-problem-with-authenticate_with_http_digest
+    - https://exchange.xforce.ibmcloud.com/vulnerabilities/51528
+    - https://github.com/advisories/GHSA-rxq3-gm4p-5fj4
+    - https://lists.apple.com/archives/security-announce/2010/Mar/msg00001.html
+    - http://support.apple.com/kb/HT4077
+    - http://n8.tumblr.com/post/117477059/security-hole-found-in-rails-2-3s
```
Comment (postmodern) [inline gems/rails/CVE-2009-2422.yml:27]: The [patch](https://gist.github.com/n8/d5fa1ce8be9dfe0d9d19) referenced in the tumblr blog post mentions `actionpack`. Should probably be moved to `gems/actionpack/`.

File: gems/rails/CVE-2008-4094.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+---
+gem: rails
+cve: 2008-4094
+ghsa: xf96-32q2-9rw2
+url: http://www.rorsecurity.info/2008/09/08/sql-injection-issue-in-limit-and-offset-parameter
```
Comment (postmodern) [inline gems/rails/CVE-2008-4094.yml:5]: The patches referenced in the [lighthouse](https://rails.lighthouseapp.com/projects/8994/tickets/288) link reference `activerecord`. Should probably be moved to `gems/activerecord/`.

File: gems/rails/CVE-2006-4112.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+unaffected_versions:
+  - "< 1.1.0"
+patched_versions:
+  - ">= 1.1.6"
+related:
+  url:
+    - https://nvd.nist.gov/vuln/detail/CVE-2006-4112
+    - https://exchange.xforce.ibmcloud.com/vulnerabilities/28364
+    - https://github.com/advisories/GHSA-9wrq-xvmp-xjc8
+
```
Comment (postmodern) [inline gems/rails/CVE-2006-4112.yml:24]: Empty line.

**Review by postmodern — CHANGES_REQUESTED** [review]
I reviewed the patches attached to the various Rails Security Release blog posts and many of the fixes are to the other rails gems (ex: `actionpack`) as opposed to the main `rails` gem itself. Some of the older Rails Security Releases didn't have patches associated with them, so not sure which rails related gem contained the fix at the time. Should probably move the files into the appropriate `gems/` sub-directories and update `gem:`.

**Comment by postmodern** [comment]
You might also want to suggest edits to the GitHub Advisory Database for those Rails Security Advisories, as they relate to specific rails gems not the [main rails gem](https://rubygems.org/gems/rails).

File: gems/rails/CVE-2008-7248.yml [hunk 7]
```diff
@@ -0,0 +1,32 @@
+---
+gem: rails
```
[the remark this answers is not in the document]
Comment (jasnow) [reply gems/rails/CVE-2008-7248.yml]: Made the changes.

File: gems/rails/CVE-2009-3009.yml (hunk 1 above)
Comment (jasnow) [reply gems/rails/CVE-2009-3009.yml]: Made the changes.

File: gems/rails/CVE-2009-3086.yml (hunk 2 above)
Comment (jasnow) [reply gems/rails/CVE-2009-3086.yml]: Made the changes.

File: gems/rails/CVE-2009-4214.yml (hunk 3 above)
Comment (jasnow) [reply gems/rails/CVE-2009-4214.yml:5]: Made the changes.

File: gems/rails/CVE-2009-2422.yml (hunk 4 above)
Comment (jasnow) [reply gems/rails/CVE-2009-2422.yml:27]: Made the changes.

File: gems/rails/CVE-2008-4094.yml (hunk 5 above)
Comment (jasnow) [reply gems/rails/CVE-2008-4094.yml]: Made the changes.

**Comment by postmodern** [comment]
Also the rails advisories will need `framework: rails`.

File: gems/actionpack/CVE-2008-7248.yml [hunk 8]
```diff
@@ -1,5 +1,5 @@
 ---
-gem: rails
+gem: actionpack
```
Comment (postmodern) [inline gems/actionpack/CVE-2008-7248.yml:2]: Will probably also need `framework: rails`, even though `bundler-audit` doesn't use it, but other downstream consumers might need it.

File: gems/actionpack/CVE-2009-3009.yml [hunk 9]
```diff
@@ -1,5 +1,5 @@
 ---
-gem: rails
+gem: actionpack
```
Comment (postmodern) [inline gems/actionpack/CVE-2009-3009.yml:2]: Here as well.

File: gems/actionpack/CVE-2009-3086.yml [hunk 10]
```diff
@@ -1,5 +1,5 @@
 ---
-gem: rails
+gem: actionpack
```
Comment (postmodern) [inline gems/actionpack/CVE-2009-3086.yml:2]: and here.

File: gems/activesupport/CVE-2009-3009.yml [hunk 11]
```diff
@@ -0,0 +1,32 @@
+---
+gem: activesupport
```
Comment (postmodern) [inline gems/activesupport/CVE-2009-3009.yml:2]: and here.

File: gems/activerecord/CVE-2008-4094.yml [hunk 12]
```diff
@@ -0,0 +1,29 @@
+---
+gem: activerecord
```
Comment (postmodern) [inline gems/activerecord/CVE-2008-4094.yml:2]: and here.

**Review by postmodern — CHANGES_REQUESTED** [review]
Probably should also add `framework: rails` since the other `actionpack` advisories also have `framework: rails`.

**Comment by jasnow** [comment]
>> "Also the rails advisories will need framework: rails.

Done

**Pull request merged**